Adding content for the new flexible consent option (#2167)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/authorization-servers/index.md

| system                                   | Whether Okta created the Scope                                                                          | Boolean   |                | False                                      |

* A consent dialog box appears depending on the values of three elements:

>   * `prompt` - a query parameter used in requests to [`/authorize`](/docs/reference/api/oidc/#authorize)

Reviewer: ```suggestion
  * `prompt` - a query parameter that is used in requests to [`/authorize`](/docs/reference/api/oidc/#authorize)
```

  * `consent` - a property on Scopes as listed in the table above. Allows you to individually enable and disable user consent for a scope.

  | `prompt` Value   | `consent_method`        | `consent`                            | Result       |
  | :--------------- | :---------------------- | :----------------------------------- | :----------- |
  | `CONSENT`        | `TRUSTED` or `REQUIRED` | `REQUIRED`                           | Prompted     |
  | `CONSENT`        | `TRUSTED` or `REQUIRED` | `FLEXIBLE`                           | Prompted     |
  | `CONSENT`        | `TRUSTED`               | `IMPLICIT`                           | Not prompted |
  | `NONE`           | `TRUSTED`               | `FLEXIBLE`, `IMPLICIT`, or `REQUIRED`| Not prompted |
  | `NONE`           | `REQUIRED`              | `FLEXIBLE` or `REQUIRED`             | Prompted     |
  | `NONE`           | `REQUIRED`              | `IMPLICIT`                           | Not prompted |

> > **Note:** If `CONSENT` is set to `FLEXIBLE`, and the scope is requested when using the Client Credentials grant flow, the scope is granted in the access token with no consent prompt. This is because there is no user involved in a 2-legged OAuth [Client Credentials](/docs/guides/implement-client-creds/overview/) grant flow.

Reviewer: ```suggestion
> **Note:** If `CONSENT` is set to `FLEXIBLE` and the scope is requested when using the Client Credentials grant flow, the scope is granted in the access token with no consent prompt. This occurs because a user isn't involved in a two-legged OAuth [Client Credentials](/docs/guides/implement-client-creds/overview/) grant flow.
```

<!-- If you change this section, change it in apps.md (/docs/reference/api/apps/#credentials-settings-details) and oidc.md (/docs/reference/api/oidc/#scopes) as well. Add 'LOGIN' to the first three rows when supported -->

**Notes:**

> * Apps created on `/api/v1/apps` default to `consent_method=TRUSTED`, while those created on `/api/v1/clients` default to `consent_method=REQUIRED`.

Reviewer 2: @yuliu-okta Is there a reason we default to `trusted` when created via /apps API

Reviewer 3: https://oktawiki.atlassian.net/wiki/spaces/PM/pages/232364836/Fed+-+OAuth+2+Consent+and+Revocation, /apps API is internal vs /clients is a protocol defined API for client registration.

> * If you request a Scope that requires consent while using the `client_credentials` flow, an error is returned. Because there is no user, no consent can be given.

Reviewer: ```suggestion
* If you request a Scope that requires consent while using the `client_credentials` flow, an error is returned. Since there is no user, consent can't be given.
```

> * If the `prompt` value is set to `NONE`, but the `consent_method` and the `consent` values are set to `REQUIRED`, then an error occurs.

Reviewer 2: Why is this? What error did you see?

Author: @venkatviswanathan-okta - this was existing content - not sure why it's appearing like it is new content (in green). Not sure what the error is. Who is in charge of the authorization servers reference api content? They may know. @yuliu-okta?

Reviewer 3: it is basically mean it will fail with `prompt=none` and consent is required because we cannot show any UI in this case. The error code should be `consent_required`

> * The Scope name must only contain printable ASCII except for spaces, double quotes, and backslashes. It also must not start with `okta.` or `okta:` and must not be only `okta` or `*`.

Reviewer: (Optional) must > can
